File: tests/hull_moving_average_test.py
import numpy as np

from taflow import HullMovingAverage


def test_lifecycle() -> None:
    values = np.linspace(1.0, 20.0, 32)
    state = HullMovingAverage(9).extend(values)
    assert len(state) == len(values)
    assert state.reset().value is None

File: crates/taflow-core/src/indicators/hull_moving_average_test.rs
use super::hull_moving_average::HullMovingAverage;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = HullMovingAverage::new(9).unwrap();
    for value in 0..12 {
        state.append(value as f64);
    }
    assert!(state.value().is_some());
    state.reset();
    assert_eq!(state.value(), None);
}
